datafusion/src/physical_plan/functions.rs

```diff
@@ -207,6 +215,14 @@ pub enum BuiltinScalarFunction {
     RegexpMatch,
 }
 
+impl BuiltinScalarFunction {
+    /// an allowlist of functions to take zero arguments, so that they will get special treatment
+    /// while executing.
+    fn supports_zero_argument(&self) -> bool {
```

Comment:
had to add this whitelist because vararg funcs like `array` will not accept 0 args